src/gui/qgsattributeeditorcontext.h

```diff
@@ -103,6 +103,14 @@ class GUI_EXPORT QgsAttributeEditorContext
       Q_ASSERT( parentContext.vectorLayerTools() );
     }
 
+    ~QgsAttributeEditorContext()
+    {
+      if ( mParentContext )
+      {
+        delete mParentContext;
```

Comment:
Thank you! 
Is it possible to use a `std::unique_ptr` instead of a raw pointer? We are trying to get rid of `new` and `delete` where possible.

Reply:
- Are we sure we don't care about parent state changes at all and can just hold a copy?
- If yes, could we store by value instead of pointer?
- If no, should we rather switch to a QObject (guarded pointer) based approach instead?

Reply:
Actually after more tests I am not sure about the first point. I've got several crashes appearing with the last commit instantiating a new QgsAttributeEditorContext as the parent so we may be stuck with the previous implementation.

Using a std::unique_ptr is not convenient here (pointer on same class type + copy constructor deletion). Also we cannot store directly a value because it's the same type (does not compile).

I may be able to spend more time to think about this later, but right now the first commit still fix the main crash.

Reply:
Totally agree! :+1: